libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SFSDKSalesforceSDKUpgradeManager.m

```diff
@@ -0,0 +1,176 @@
+//  are permitted provided that the following conditions are met:
+//  * Redistributions of source code must retain the above copyright notice, this list of conditions
+//  and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright notice, this list of
+//  conditions and the following disclaimer in the documentation and/or other materials provided
+//  with the distribution.
+//  * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+//  endorse or promote products derived from this software without specific prior written
+//  permission of salesforce.com, inc.
+// 
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+//  IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+//  FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+//  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+//  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+//  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+//  WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#import <SalesforceSDKCommon/NSUserDefaults+SFAdditions.h>
+#import <SalesforceSDKCore/SalesforceSDKCore-Swift.h>
+#import "SFSDKSalesforceSDKUpgradeManager.h"
+#import "SFDirectoryManager+Internal.h"
+#import "SFUserAccount+Internal.h"
+#import "SFKeyStoreManager.h"
+#import "SFDefaultUserAccountPersister.h"
+
+NSString * const kSalesforceSDKManagerVersionKey = @"com.salesforce.mobilesdk.salesforcesdkmanager.version";
+
+@implementation SFSDKSalesforceSDKUpgradeManager
+
++ (void)upgrade {
+    NSString *lastVersion = [SFSDKSalesforceSDKUpgradeManager lastVersion];
+    NSString *currentVersion = [SFSDKSalesforceSDKUpgradeManager currentVersion];
+    
+    if ([currentVersion isEqualToString:lastVersion]) {
+        return;
+    }
+    
+    if (!lastVersion || [lastVersion doubleValue] < 9.2) {
+        [SFDirectoryManager upgradeUserDirectories];
+        [SFSDKSalesforceSDKUpgradeManager upgradeUserAccounts];
+        [NSURLCache.sharedURLCache removeAllCachedResponses]; // For cache encryption key change
+    }
+    
+    [[NSUserDefaults msdkUserDefaults] setValue:currentVersion forKey:kSalesforceSDKManagerVersionKey];
+    [[NSUserDefaults msdkUserDefaults] synchronize];
+}
+
++ (NSString *)lastVersion {
+    return [[NSUserDefaults msdkUserDefaults] stringForKey:kSalesforceSDKManagerVersionKey];
+}
+
++ (NSString *)currentVersion {
+    return [[[NSBundle bundleForClass:[SFSDKSalesforceSDKUpgradeManager class]] infoDictionary] objectForKey:@"CFBundleShortVersionString"];
```

Comment:
Will this be the SDK version or the app version? If it's the app version, it won't work reliably.

Reply:
SDK version

Reply:
👍🏻